src/utils/userLocation.ts

```diff
@@ -6,99 +6,29 @@ import { useInterval } from "hooks/useInterval";
 
 import { updateUserProfile } from "pages/Account/helpers";
 
-import { logEventGoogleAnalytics } from "./googleAnalytics";
 import { getCurrentTimeInMilliseconds } from "./time";
-import { openRoomUrl } from "./url";
 
-export type LocationData = Record<string, number>;
+export type LocationData = string | null;
 
 export interface UpdateLocationDataProps {
   userId: string;
-  newLocationData: LocationData;
+  newLocationPath: LocationData;
 }
 
 export const updateLocationData = ({
   userId,
-  newLocationData,
+  newLocationPath,
 }: UpdateLocationDataProps) => {
   updateUserProfile(userId, {
     lastSeenAt: getCurrentTimeInMilliseconds(),
-    lastSeenIn: newLocationData,
-  });
-};
-
-export interface SetLocationDataProps {
-  userId: string;
-  locationName: string;
-}
-
-// TODO: refactor how user location updates works here?
-//   Called from VenuePage useEffect when venue changes, etc
-export const setLocationData = ({
-  userId,
-  locationName,
-}: SetLocationDataProps) => {
-  updateLocationData({
-    userId,
-    newLocationData: {
-      [locationName]: getCurrentTimeInMilliseconds(),
-    },
-  });
-};
-
-export interface UpdateCurrentLocationDataProps {
-  userId: string;
-  profileLocationData: LocationData;
-}
-
-// NOTE: The intended effect is to update the current location, without rewriting it.
-// profileLocationData can only have 1 key at any point of time
-// TODO: refactor how user location updates works here?
-//   Called from VenuePage interval
-export const updateCurrentLocationData = ({
-  userId,
-  profileLocationData,
-}: UpdateCurrentLocationDataProps) => {
-  const [locationName] = Object.keys(profileLocationData);
-
-  updateLocationData({
-    userId,
-    newLocationData: { [locationName]: getCurrentTimeInMilliseconds() },
+    lastSeenIn: newLocationPath,
```

Comment:
Should this renamed to the new field name?

Reply:
Good catch. It was because of the type inconsistencies. Fixed in https://github.com/sparkletown/sparkle/pull/2202/commits/1c84be82101f89a5bfdb27401c60043d07c0999f